### PR TITLE
Pillar rich links

### PR DESCRIPTION
**Pull request by webb04**
## Why are you doing this?

- Enhance rich links client side
- Match the designs on dotcom
- Give readers more information about rich links (e.g. what pillar/topic)
- Make them more appealing to drive onward content

## Changes

- Accept new `richlink` query parameter to serve article details (pillar and image) and without html document, scripts and styles
- Request article details for rich links client side
- New `darkModeStyles` function that returns a string of dark mode styles instead of a `SerializedStyles` object
- Move `RichLink` component into separate file
- Add `RichLink` story

## Other considerations

Creating a new endpoint saves us having to send large unused html, css and js. It does split the cache though and will put extra load on MAPI and apps-rendering.

## Screenshots

| Before | After |
| --- | --- |
|  |  |
|  |  |

### CODE REVIEW

File: src/components/richLink.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { text, withKnobs } from '@storybook/addon-knobs';
+import type { FC } from 'react';
+import React from 'react';
+import { basePx } from 'styles';
+import { selectPillar } from '../storybookHelpers';
+import RichLink, { richLinkWidth } from './richLink';
+
+const overrideStyle = css`
+	${from.wide} {
+		margin-left: calc(${richLinkWidth} + ${basePx(2)} + ${basePx(3)});
```
Comment (dskamiotis) [inline src/components/richLink.stories.tsx:13]: nice, looks smart would like a walk through if poss 🎉

File: src/components/richLink.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+const richLinkStyles = (format: Format): SerializedStyles => {
+	const backgroundColor =
+		format.design === Design.Comment ? neutral[86] : neutral[97];
+
+	const { kicker: newsKicker } = getThemeStyles(Pillar.News);
+	const { kicker: opinionKicker } = getThemeStyles(Pillar.Opinion);
+	const { kicker: sportKicker } = getThemeStyles(Pillar.Sport);
+	const { kicker: cultureKicker } = getThemeStyles(Pillar.Culture);
+	const { kicker: lifestyleKicker } = getThemeStyles(Pillar.Lifestyle);
```
Comment (dskamiotis) [inline src/components/richLink.tsx:53]: updated from getPillar?

**Review by dskamiotis — APPROVED** [review]

File: src/components/richLink.stories.tsx (hunk 1 above)
Comment (webb04) [reply src/components/richLink.stories.tsx:13]: The rich link component has a negative margin so it goes offscreen in storybook, so this is the opposite margin to bring it back on screen 😅

File: src/components/richLink.tsx (hunk 2 above)
Comment (webb04) [reply src/components/richLink.tsx:53]: yep we renamed `getPillar` to `getTheme` recently

File: src/server/server.ts [hunk 3]
```diff
@@ -136,15 +139,44 @@ async function serveArticle(
...
+
+	if (
+		item.mainMedia.kind === OptionKind.Some &&
+		item.mainMedia.value.kind === MainMediaKind.Image
+	) {
+		res.set('image', item.mainMedia.value.image.src);
+	}
+
+	res.set('pillar', renderingRequest.content.pillarName);
+	res.status(200).end();
```
Comment (JamieB-gu) [inline src/server/server.ts:163]: Is your expectation that this returns an empty response with these two headers added? In PROD (after modifying the url to include `rendered-items`) I'm seeing a response that instead returns HTML _without_ these headers 🤔

File: src/client/article.ts [hunk 4]
```diff
@@ -321,6 +321,31 @@ function localDates(): void {
 	}
 }
 
+function richLinks(): void {
+	document
+		.querySelectorAll('.js-rich-link[data-article-id]')
+		.forEach((richLink) => {
+			const articleId = richLink.getAttribute('data-article-id');
+			if (articleId) {
+				void fetch(`${articleId}?richlink`).then((response) => {
```
[the remark this answers is not in the document]
Comment (webb04) [reply src/client/article.ts:330]: Realised today this was only working locally and not in production. These two PRs should hopefully fix it and is a bit of the work to be able to use editions in prod
- https://github.com/guardian/apps-rendering/pull/943
- https://github.com/guardian/mobile-apps-api/pull/1661

**Pull request merged**